File: apps/s-2-admin/src/api/appModel/AppModel.ts
import { JsonValue } from "type-fest";
import { Credential } from "../credential/Credential";
import { Webhook } from "../webhook/Webhook";
import { ApiKey } from "../apiKey/ApiKey";

export type AppModel = {
  id: string;
  dirName: string;
  keys: JsonValue;
  categories?: Array<
    "calendar" | "messaging" | "other" | "payment" | "video" | "web3"
  >;
  createdAt: Date;
  updatedAt: Date;
  credentials?: Array<Credential>;
  webhook?: Array<Webhook>;
  apiKey?: Array<ApiKey>;
};

File: apps/cactus-demo/src/Infrastructure/Models/AppModel.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using CactusDemo.Core.Enums;

namespace CactusDemo.Infrastructure.Models;

[Table("AppModels")]
public class AppModelDbModel
{
    [Key()]
    [Required()]
    public string Id { get; set; }

    [Required()]
    [StringLength(256)]
    public string DirName { get; set; }

    public string? Keys { get; set; }

    [Required()]
    public CategoriesEnum Categories { get; set; } = CategoriesEnum.calendar;

    [Required()]
    public DateTime CreatedAt { get; set; }

    [Required()]
    public DateTime UpdatedAt { get; set; }

    public List<ApiKeyDbModel>? ApiKey { get; set; } = new List<ApiKeyDbModel>();

    public List<CredentialDbModel>? Credential { get; set; } = new List<CredentialDbModel>();

    public List<WebhookDbModel>? Webhook { get; set; } = new List<WebhookDbModel>();
}
